xds/src/main/java/io/grpc/xds/orca/OrcaServiceImpl.java

```diff
@@ -36,26 +39,51 @@
 
 /**
  * Implements a {@link BindableService} that generates Out-Of-Band server metrics.
+ * Register the returned service to the server, then a client can request for periodic load reports.
  */
 public final class OrcaServiceImpl implements BindableService {
   private static final Logger logger = Logger.getLogger(OrcaServiceImpl.class.getName());
 
+  /**
+   * Empty or invalid (non-positive) minInterval config in will be treated to this default value.
+   */
+  public static final long DEFAULT_MIN_REPORT_INTERVAL_NANOS = TimeUnit.SECONDS.toNanos(30);
+
   private final long minReportIntervalNanos;
   private final ScheduledExecutorService timeService;
   @VisibleForTesting
   final AtomicInteger clientCount = new AtomicInteger(0);
-  private OrcaMetrics orcaMetrics;
+  private MetricRecorder metricRecorder;
   private final RealOrcaServiceImpl delegate = new RealOrcaServiceImpl();
 
   /**
    * Constructs a service to report server metrics. Config the report interval lower bound, the
-   * executor to run the timer, and a {@link OrcaMetrics} that contains metrics data.
+   * executor to run the timer, and a {@link MetricRecorder} that contains metrics data.
+   *
+   * @param minInterval configures the minimum metrics reporting interval for the
+   *        service. Bad configuration (non-positive) will be overridden to service default (30s).
+   *        Minimum metrics reporting interval means, if the setting in the client's
+   *        request is invalid (non-positive) or below this value, they will be treated
+   *        as this value.
    */
-  public OrcaServiceImpl(long minReportIntervalNanos, ScheduledExecutorService timeService,
-                         OrcaMetrics orcaMetrics) {
-    this.minReportIntervalNanos = minReportIntervalNanos;
+  public static BindableService createService(long minInterval, TimeUnit timeUnit,
```

Comment:
Let's have the addition arguments here (minInterval, timeUnit) placed at the end of the argument list, so that it more closely mirrors the two-arg `createService(ScheduledExecutorService, MetricRecorder)`.